# Imports

In [1]:
import pandas as pd
import matplotlib.pyplot as plt
import pandas as pd
import seaborn as sns

import nltk

import warnings
warnings.filterwarnings("ignore")

import prepare as p
import explore as e

# Acquire

In [2]:
df = pd.read_excel('survey_data.xlsx')
df.head()

respondent_id  collector_id               start                 end    zip  \
0    12488725288     399921329 2021-03-13 12:23:16 2021-03-13 12:31:36  20124   
1      124886789     399921329 2021-03-13 11:48:18 2021-03-13 11:51:46  22191   
2      124884534     399921329 2021-03-13 09:25:30 2021-03-13 09:37:07  22181   
3    12488374327     399921329 2021-03-13 08:04:15 2021-03-13 08:19:41  22601   
4    12488363511     399921329 2021-03-13 07:56:08 2021-03-13 08:00:58  22304   

   q01_overall                                   q01_overall_open  \
0          1.0                                                NaN   
1          2.0                                                NaN   
2          1.0                                                NaN   
3          1.0  Covid doesn't scare me; I think masks are unne...   
4          2.0                                                NaN   

   q02_financial q02_financial_open  \
0            1.0                NaN   
1            2.0                NaN   
2            1.0                NaN   
3            1.0                NaN   
4            1.0          No impact   

                                  q03_pregnancy_open  ... q06_support_none  \
0  Understanding transmission of COVID 19 and how...  ...              NaN   
1  How the vaccine affects both pregnancy and pos...  ...              NaN   
2                     Get Covid and be hospitalized.  ...              NaN   
3  I hate all of the restrictions--I had to quara...  ...              NaN   
4  Becoming sick with COVID and being unable to c...  ...              NaN   

  q06_support_other q08_hispanic q09_gender  \
0               NaN           No      Woman   
1               NaN           No      Woman   
2               NaN          Yes      Woman   
3               NaN           No      Woman   
4               NaN           No      Woman   

                                            q11_open duration pregnancy_stage  \
0  The support we had while in the maternity ward...       -9          1 year   
1                                                NaN       -4   3rd trimester   
2  Because of COVID, my husband hasn�t been avail...      -12   3rd trimester   
3                                                NaN      -16          1 year   
4                                                NaN       -5          1 year   

  miscarriage   race             income  
0       False  White   125,001 and over  
1       False  Black  100,001 - 125,000  
2       False  White  100,001 - 125,000  
3       False  White    40,001 - 75,000  
4       False  White   125,001 and over  

[5 rows x 37 columns]

In [3]:
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 712 entries, 0 to 711
Data columns (total 37 columns):
 #   Column                Non-Null Count  Dtype         
---  ------                --------------  -----         
 0   respondent_id         712 non-null    int64         
 1   collector_id          712 non-null    int64         
 2   start                 712 non-null    datetime64[ns]
 3   end                   712 non-null    datetime64[ns]
 4   zip                   698 non-null    object        
 5   q01_overall           690 non-null    float64       
 6   q01_overall_open      59 non-null     object        
 7   q02_financial         687 non-null    float64       
 8   q02_financial_open    49 non-null     object        
 9   q03_pregnancy_open    655 non-null    object        
 10  q04_feeding_open      616 non-null    object        
 11  q05_needs_1st         469 non-null    object        
 12  q05_needs_2nd         368 non-null    object        
 13  q05_needs_3rd       

In [4]:
df.describe()

respondent_id  collector_id  q01_overall  q02_financial      duration
count   7.120000e+02  7.120000e+02   690.000000     687.000000    712.000000
mean    5.848256e+09  3.014783e+08     2.385507       1.852984    -34.675562
std     6.156543e+09  1.712275e+08     0.937414       1.000110    506.979140
min     1.232400e+04  3.997714e+06     1.000000       1.000000 -13391.000000
25%     1.239695e+08  3.998559e+08     2.000000       1.000000     -9.000000
50%     1.248721e+08  3.998559e+08     2.000000       2.000000     -6.000000
75%     1.241387e+10  3.999213e+08     3.000000       2.000000     -4.000000
max     1.248873e+10  3.999213e+08     5.000000       5.000000     -1.000000

In [5]:
df.columns

Index(['respondent_id', 'collector_id', 'start', 'end', 'zip', 'q01_overall',
       'q01_overall_open', 'q02_financial', 'q02_financial_open',
       'q03_pregnancy_open', 'q04_feeding_open', 'q05_needs_1st',
       'q05_needs_2nd', 'q05_needs_3rd', 'q06_birthdoula',
       'q06_postpartumdoula', 'q06_lactation', 'q06_mentalhealth',
       'q06_social', 'q06_diapers', 'q06_formula', 'q06_food', 'q06_housing',
       'q06_transportation', 'q06_healthinsurance', 'q06_financial',
       'q06_covid19education', 'q06_support_none', 'q06_support_other',
       'q08_hispanic', 'q09_gender', 'q11_open', 'duration', 'pregnancy_stage',
       'miscarriage', 'race', 'income'],
      dtype='object')

# Prepare

In [6]:
# drop non-quantitative and irrelivent data
df = df[['race', 'q08_hispanic', 'income', 'pregnancy_stage', 'q01_overall',
       'q02_financial','q06_birthdoula','q06_postpartumdoula', 
       'q06_lactation', 'q06_mentalhealth', 'q06_social', 
       'q06_diapers', 'q06_formula', 'q06_food', 
       'q06_housing','q06_transportation', 'q06_healthinsurance', 
       'q06_financial','q06_covid19education', 'q06_support_none', 
       'q06_support_other']]
df.head()

race q08_hispanic             income pregnancy_stage  q01_overall  \
0  White           No   125,001 and over          1 year          1.0   
1  Black           No  100,001 - 125,000   3rd trimester          2.0   
2  White          Yes  100,001 - 125,000   3rd trimester          1.0   
3  White           No    40,001 - 75,000          1 year          1.0   
4  White           No   125,001 and over          1 year          2.0   

   q02_financial q06_birthdoula       q06_postpartumdoula  \
0            1.0            NaN  Postpartum Doula Support   
1            2.0            NaN                       NaN   
2            1.0            NaN  Postpartum Doula Support   
3            1.0            NaN                       NaN   
4            1.0            NaN  Postpartum Doula Support   

                        q06_lactation       q06_mentalhealth  ... q06_diapers  \
0  Breastfeeding or Lactation Support                    NaN  ...         NaN   
1  Breastfeeding or Lactation Support                    NaN  ...         NaN   
2  Breastfeeding or Lactation Support  Mental Health Support  ...         NaN   
3  Breastfeeding or Lactation Support  Mental Health Support  ...         NaN   
4                                 NaN                    NaN  ...         NaN   

  q06_formula q06_food q06_housing q06_transportation  \
0         NaN      NaN         NaN                NaN   
1         NaN      NaN         NaN                NaN   
2         NaN      NaN         NaN                NaN   
3         NaN      NaN         NaN                NaN   
4         NaN      NaN         NaN                NaN   

         q06_healthinsurance q06_financial  \
0  Health Insurance Coverage           NaN   
1                        NaN           NaN   
2                        NaN           NaN   
3                        NaN           NaN   
4                        NaN           NaN   

                                q06_covid19education q06_support_none  \
0  Education/Informational Resources related to C...              NaN   
1  Education/Informational Resources related to C...              NaN   
2  Education/Informational Resources related to C...              NaN   
3                                                NaN              NaN   
4                                                NaN              NaN   

  q06_support_other  
0               NaN  
1               NaN  
2               NaN  
3               NaN  
4               NaN  

[5 rows x 21 columns]

In [7]:
# change data types of overall and financial to objects
df = df.astype(object)
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 712 entries, 0 to 711
Data columns (total 21 columns):
 #   Column                Non-Null Count  Dtype 
---  ------                --------------  ----- 
 0   race                  712 non-null    object
 1   q08_hispanic          625 non-null    object
 2   income                616 non-null    object
 3   pregnancy_stage       712 non-null    object
 4   q01_overall           690 non-null    object
 5   q02_financial         687 non-null    object
 6   q06_birthdoula        153 non-null    object
 7   q06_postpartumdoula   254 non-null    object
 8   q06_lactation         271 non-null    object
 9   q06_mentalhealth      375 non-null    object
 10  q06_social            487 non-null    object
 11  q06_diapers           55 non-null     object
 12  q06_formula           43 non-null     object
 13  q06_food              45 non-null     object
 14  q06_housing           20 non-null     object
 15  q06_transportation    9 non-null      ob

In [8]:
# change responces of overall and to catagories

df.q01_overall = df.q01_overall.apply(lambda number : str(number).replace('1.0','We\'ve got this!'))
df.q01_overall = df.q01_overall.apply(lambda number : str(number).replace('2.0','We\'re doing OK.'))
df.q01_overall = df.q01_overall.apply(lambda number : str(number).replace('3.0','So-So.'))
df.q01_overall = df.q01_overall.apply(lambda number : str(number).replace('4.0','It\'s a struggle.'))
df.q01_overall = df.q01_overall.apply(lambda number : str(number).replace('5.0','Barely making it.'))

df.q02_financial = df.q02_financial.apply(lambda number : str(number).replace('1.0','We\'ve got this!'))
df.q02_financial = df.q02_financial.apply(lambda number : str(number).replace('2.0','We\'re doing OK.'))
df.q02_financial = df.q02_financial.apply(lambda number : str(number).replace('3.0','So-So.'))
df.q02_financial = df.q02_financial.apply(lambda number : str(number).replace('4.0','It\'s a struggle.'))
df.q02_financial = df.q02_financial.apply(lambda number : str(number).replace('5.0','Barely making it.'))
                                                                          
print(df.q01_overall.value_counts())
print('~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~')
print(df.q02_financial.value_counts())

We're doing OK.      327
So-So.               154
We've got this!      104
It's a struggle.      99
nan                   22
Barely making it.      6
Name: q01_overall, dtype: int64
~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
We've got this!      314
We're doing OK.      235
So-So.                75
It's a struggle.      51
nan                   25
Barely making it.     12
Name: q02_financial, dtype: int64


In [9]:
# fixing value mismatch 

df.income = df.income.apply(lambda number : str(number).replace('$40,000 or under','40,000 and under'))

df.income.value_counts()

125,001 and over     238
75,001 - 100,000     115
100,001 - 125,000    105
40,001 - 75,000      104
nan                   96
40,000 and under      54
Name: income, dtype: int64

In [10]:
# Replace nans with "Zero Response"
df = df.fillna(value="Zero Response")
df.head()

race q08_hispanic             income pregnancy_stage      q01_overall  \
0  White           No   125,001 and over          1 year  We've got this!   
1  Black           No  100,001 - 125,000   3rd trimester  We're doing OK.   
2  White          Yes  100,001 - 125,000   3rd trimester  We've got this!   
3  White           No    40,001 - 75,000          1 year  We've got this!   
4  White           No   125,001 and over          1 year  We're doing OK.   

     q02_financial q06_birthdoula       q06_postpartumdoula  \
0  We've got this!  Zero Response  Postpartum Doula Support   
1  We're doing OK.  Zero Response             Zero Response   
2  We've got this!  Zero Response  Postpartum Doula Support   
3  We've got this!  Zero Response             Zero Response   
4  We've got this!  Zero Response  Postpartum Doula Support   

                        q06_lactation       q06_mentalhealth  ...  \
0  Breastfeeding or Lactation Support          Zero Response  ...   
1  Breastfeeding or Lactation Support          Zero Response  ...   
2  Breastfeeding or Lactation Support  Mental Health Support  ...   
3  Breastfeeding or Lactation Support  Mental Health Support  ...   
4                       Zero Response          Zero Response  ...   

     q06_diapers    q06_formula       q06_food    q06_housing  \
0  Zero Response  Zero Response  Zero Response  Zero Response   
1  Zero Response  Zero Response  Zero Response  Zero Response   
2  Zero Response  Zero Response  Zero Response  Zero Response   
3  Zero Response  Zero Response  Zero Response  Zero Response   
4  Zero Response  Zero Response  Zero Response  Zero Response   

  q06_transportation        q06_healthinsurance  q06_financial  \
0      Zero Response  Health Insurance Coverage  Zero Response   
1      Zero Response              Zero Response  Zero Response   
2      Zero Response              Zero Response  Zero Response   
3      Zero Response              Zero Response  Zero Response   
4      Zero Response              Zero Response  Zero Response   

                                q06_covid19education q06_support_none  \
0  Education/Informational Resources related to C...    Zero Response   
1  Education/Informational Resources related to C...    Zero Response   
2  Education/Informational Resources related to C...    Zero Response   
3                                      Zero Response    Zero Response   
4                                      Zero Response    Zero Response   

  q06_support_other  
0     Zero Response  
1     Zero Response  
2     Zero Response  
3     Zero Response  
4     Zero Response  

[5 rows x 21 columns]

In [11]:
df.income.value_counts()

125,001 and over     238
75,001 - 100,000     115
100,001 - 125,000    105
40,001 - 75,000      104
nan                   96
40,000 and under      54
Name: income, dtype: int64

In [12]:
# create non-white column for respondance who identify as non-white
df.nonwhite = (df.race != "White") & (df.race != "Undisclosed")

df.nonwhite.value_counts()

False    602
True     110
Name: race, dtype: int64

In [13]:
df.race.value_counts()

White                 512
Undisclosed            90
Black                  41
Asian                  36
Multiracial            21
Hispanic or Latino      6
Pacific Islander        3
American Indian         3
Name: race, dtype: int64

In [14]:
df.q08_hispanic.value_counts()

No               585
Zero Response     87
Yes               40
Name: q08_hispanic, dtype: int64

# Exploration

## Looking at the whole dataset

Breakdown of race

In [15]:
df.race.value_counts(normalize=True)

White                 0.719101
Undisclosed           0.126404
Black                 0.057584
Asian                 0.050562
Multiracial           0.029494
Hispanic or Latino    0.008427
Pacific Islander      0.004213
American Indian       0.004213
Name: race, dtype: float64

### Overall Race
* White 72%
* Black 6%
* Asian 5%
* Multiracial 3%
* Undisclosed 13%
* Other 1%

In [16]:
df.q08_hispanic.value_counts(normalize=True)

No               0.821629
Zero Response    0.122191
Yes              0.056180
Name: q08_hispanic, dtype: float64

### Overall Hispanic
* 6%

In [17]:
df.income.value_counts(normalize=True)

125,001 and over     0.334270
75,001 - 100,000     0.161517
100,001 - 125,000    0.147472
40,001 - 75,000      0.146067
nan                  0.134831
40,000 and under     0.075843
Name: income, dtype: float64

### Overall Household Income
* Over 125,000  33%
* Between 100,001 and 125,000 15%
* Between 75,001 and 100,000 16%
* Between 40,001 and 75,000 15% 
* Between 40,000 and under 08%
* Undisclosed  13%

In [18]:
df.q01_overall.value_counts(normalize=True)

We're doing OK.      0.459270
So-So.               0.216292
We've got this!      0.146067
It's a struggle.     0.139045
nan                  0.030899
Barely making it.    0.008427
Name: q01_overall, dtype: float64

### Overall Wellness 
* We've got this! 15%
* We're doing OK. 46%
* So-so. 22%
* It's a struggle. 14%
* Bearly making it. 1%
* Undisclosed 3%

In [19]:
df.q02_financial.value_counts(normalize=True)

We've got this!      0.441011
We're doing OK.      0.330056
So-So.               0.105337
It's a struggle.     0.071629
nan                  0.035112
Barely making it.    0.016854
Name: q02_financial, dtype: float64

## Income Wellness
* We've got this! 44%
* We're doing OK. 33%
* So-so. 11%
* It's a struggle. 7%
* Bearly making it. 2%
* Undisclosed 6%

In [20]:
li = ['q06_social', 'q06_mentalhealth', 'q06_lactation','q06_postpartumdoula','q06_birthdoula',
      'q06_covid19education', 'q06_financial', 'q06_healthinsurance', 'q06_diapers', 'q06_food',
      'q06_formula', 'q06_support_none', 'q06_housing', 'q06_transportation']



print("Number of Times Support Was Identified as Needed")
print('~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~')

num = 1

for col in li:
    
    print(col[4:])
    print(df[df[col] != "Zero Response"][col].count())
    print('~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~')
    
    num += 1
    

Number of Times Support Was Identified as Needed
~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
social
487
~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
mentalhealth
375
~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
lactation
271
~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
postpartumdoula
254
~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
birthdoula
153
~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
covid19education
151
~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
financial
117
~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
healthinsurance
72
~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
diapers
55
~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
food
45
~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
formula
43
~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
support_none
39
~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
housing
20
~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
transportation
9
~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~

In [21]:
print("Percent of Respondants who Identified this Support as a Need")
print('~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~')

num = 1

for col in li:
    
    print(f'{num})',col[4:],round(df[df[col] != "Zero Response"][col].count()/len(df),2))
    
    num += 1

Percent of Respondants who Identified this Support as a Need
~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
1) social 0.68
2) mentalhealth 0.53
3) lactation 0.38
4) postpartumdoula 0.36
5) birthdoula 0.21
6) covid19education 0.21
7) financial 0.16
8) healthinsurance 0.1
9) diapers 0.08
10) food 0.06
11) formula 0.06
12) support_none 0.05
13) housing 0.03
14) transportation 0.01


In [22]:
print("Support Needed by Rank")
print('~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~')

num = 1

for col in li:
    
    print(f'{num})',col[4:])
    num += 1
    

Support Needed by Rank
~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
1) social
2) mentalhealth
3) lactation
4) postpartumdoula
5) birthdoula
6) covid19education
7) financial
8) healthinsurance
9) diapers
10) food
11) formula
12) support_none
13) housing
14) transportation


In [23]:
len(df)

712

# Overall Persona

### Respondants
* 712

### Race / Ethnicity
* White 72%
* Black 6%
* Asian 5%
* Multiracial 3%
* Undisclosed 13%
* Other 1%
<br><br>
* Hispanic 6%

### Neighborhood

### Household Income
* Over 125,000  33%
* Between 100,001 and 125,000 15%
* Between 75,001 and 100,000 16%
* Between 40,001 and 75,000 15% 
* Between 40,000 and under 08%
* Undisclosed  13%

### Wellness 
* We've got this! 15%
* We're doing OK. 46%
* So-so. 22%
* It's a struggle. 14%
* Bearly making it. 1%
* Undisclosed 3%

### Percent of Respondants Identifying need by Rank
1) social 0.68<br>
2) mentalhealth 0.53<br>
3) lactation 0.38<br>
4) postpartumdoula 0.36<br>
5) financial 0.16<br>
6) birthdoula 0.21<br>
7) diapers 0.08<br>
8) food 0.06<br>
9) covid19education 0.21<br>
10) healthinsurance 0.1<br>
11) formula 0.06<br>
12) housing 0.03<br>
13) transportation 0.01<br>
14) support_none 0.05<br>

# Looking at Persona's by Race

#### Proceeding only with values we have representitive samples of. Using 25 as a cutoff.

In [24]:
df.race.value_counts()

White                 512
Undisclosed            90
Black                  41
Asian                  36
Multiracial            21
Hispanic or Latino      6
Pacific Islander        3
American Indian         3
Name: race, dtype: int64

#### Proceeding with White, Black, Asian, and non_white

In [25]:
# preparing data frames

df_white = df[df.race=='White']
df_black = df[df.race=='Black']
df_asian = df[df.race=='Asian']
df_nonwhite = df[df.nonwhite == True]

print(df_white.race.value_counts())
print('~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~')
print(df_black.race.value_counts())
print('~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~')
print(df_asian.race.value_counts())
print('~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~')
print(df_nonwhite.race.value_counts())
print('~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~')

White    512
Name: race, dtype: int64
~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
Black    41
Name: race, dtype: int64
~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
Asian    36
Name: race, dtype: int64
~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
Black                 41
Asian                 36
Multiracial           21
Hispanic or Latino     6
Pacific Islander       3
American Indian        3
Name: race, dtype: int64
~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~


In [26]:
# list of dataframes and titles
df_list = [df_white,df_black,df_asian,df_nonwhite]
title_list = ['White','Black','Asian','Non_White']

## Respondants

In [27]:
num = 0
for df1 in df_list:
    
    print(title_list[num] + ": Number of Respondants")
    print(len(df1))
    print('~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~')
    num += 1

White: Number of Respondants
512
~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
Black: Number of Respondants
41
~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
Asian: Number of Respondants
36
~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
Non_White: Number of Respondants
110
~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~


## Wellness Distribution

In [28]:
num = 0
for df1 in df_list:
    
    print(title_list[num] + ": Wellness Distribution")
    print(df1.q01_overall.value_counts(normalize=True))
    print('~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~')
    num += 1

White: Wellness Distribution
We're doing OK.      0.464844
So-So.               0.228516
It's a struggle.     0.156250
We've got this!      0.142578
Barely making it.    0.007812
Name: q01_overall, dtype: float64
~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
Black: Wellness Distribution
We're doing OK.     0.487805
We've got this!     0.195122
It's a struggle.    0.170732
So-So.              0.146341
Name: q01_overall, dtype: float64
~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
Asian: Wellness Distribution
We're doing OK.      0.527778
So-So.               0.194444
We've got this!      0.138889
It's a struggle.     0.111111
Barely making it.    0.027778
Name: q01_overall, dtype: float64
~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
Non_White: Wellness Distribution
We're doing OK.      0.518182
So-So.               0.172727
We've got this!      0.172727
It's a struggle.     0.118182
Barely making it.    0.018182
Name:

## Household Income

In [29]:
num = 0
for df1 in df_list:
    
    print(title_list[num] + ": Income Distribution")
    print(df1.income.value_counts(normalize=True))
    print('~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~')
    num += 1

White: Income Distribution
125,001 and over     0.392578
75,001 - 100,000     0.183594
100,001 - 125,000    0.179688
40,001 - 75,000      0.160156
40,000 and under     0.068359
nan                  0.015625
Name: income, dtype: float64
~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
Black: Income Distribution
40,000 and under     0.292683
40,001 - 75,000      0.195122
75,001 - 100,000     0.170732
100,001 - 125,000    0.170732
125,001 and over     0.170732
Name: income, dtype: float64
~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
Asian: Income Distribution
125,001 and over     0.500000
40,001 - 75,000      0.222222
75,001 - 100,000     0.166667
100,001 - 125,000    0.055556
40,000 and under     0.027778
nan                  0.027778
Name: income, dtype: float64
~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
Non_White: Income Distribution
125,001 and over     0.318182
75,001 - 100,000     0.190909
40,001 - 75,000      0.1

## Perception of Financial Wellness 

In [30]:
num = 0
for df1 in df_list:
    
    print(title_list[num] + ": Perception of Financial Wellness")
    print(df1.q02_financial.value_counts(normalize=True))
    print('~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~')
    num += 1

White: Perception of Financial Wellness
We've got this!      0.482422
We're doing OK.      0.339844
So-So.               0.093750
It's a struggle.     0.066406
Barely making it.    0.017578
Name: q02_financial, dtype: float64
~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
Black: Perception of Financial Wellness
We've got this!      0.317073
We're doing OK.      0.317073
So-So.               0.170732
It's a struggle.     0.146341
Barely making it.    0.048780
Name: q02_financial, dtype: float64
~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
Asian: Perception of Financial Wellness
We're doing OK.      0.444444
We've got this!      0.305556
So-So.               0.166667
It's a struggle.     0.055556
Barely making it.    0.027778
Name: q02_financial, dtype: float64
~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
Non_White: Perception of Financial Wellness
We're doing OK.      0.354545
We've got this!      0.345455
So-So.     

## Percent of Respondance who Identified Need by Rank

In [31]:
col_list = ['q06_social', 'q06_mentalhealth', 'q06_lactation','q06_postpartumdoula','q06_birthdoula',
            'q06_covid19education', 'q06_financial', 'q06_healthinsurance', 'q06_diapers', 'q06_food',
            'q06_formula', 'q06_support_none', 'q06_housing', 'q06_transportation']

index = 0 

for df1 in df_list:
    
    print(title_list[index] + ": Percent of Respondance who Identified Need by Rank")
    
    num = 1

    for col in col_list:

        print(f'{num})',col[4:],round(df1[df1[col] != "Zero Response"][col].count()/len(df1),2))

        num += 1
        
    print('~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~')
    print()
    
    index += 1

White: Percent of Respondance who Identified Need by Rank
1) social 0.79
2) mentalhealth 0.62
3) lactation 0.42
4) postpartumdoula 0.39
5) birthdoula 0.24
6) covid19education 0.24
7) financial 0.16
8) healthinsurance 0.1
9) diapers 0.06
10) food 0.04
11) formula 0.05
12) support_none 0.07
13) housing 0.02
14) transportation 0.01
~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~

Black: Percent of Respondance who Identified Need by Rank
1) social 0.66
2) mentalhealth 0.49
3) lactation 0.44
4) postpartumdoula 0.37
5) birthdoula 0.22
6) covid19education 0.17
7) financial 0.32
8) healthinsurance 0.12
9) diapers 0.22
10) food 0.2
11) formula 0.12
12) support_none 0.0
13) housing 0.07
14) transportation 0.02
~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~

Asian: Percent of Respondance who Identified Need by Rank
1) social 0.75
2) mentalhealth 0.47
3) lactation 0.53
4) postpartumdoula 0.47
5) birthdoula 0.25
6) covid19educat

# Persona: White

## Respondants
* 512


## Niegborhood

## Household Income

* 125,001 and over     39%
* 100,001 - 125,000    18%
* 75,001 - 100,000     18%
* 40,001 - 75,000      16%
* 40,000 and under     7%
* undisclosed          2%

## Perception of Wellness
* We've got this!      14%
* We're doing OK.      46%
* So-So.               23%
* It's a struggle.     16%
* Barely making it.    1%



## Perception of Financial Wellness
* We've got this!      48%
* We're doing OK.      34%
* So-So.               9%
* It's a struggle.     7%
* Barely making it.    2%


## Hierarachy of Needs
1) social 79%<br>
2) mentalhealth 62%<br>
3) lactation 42%<br>
4) postpartumdoula 39%<br>
5) birthdoula 24%<br>
6) covid19education 24%<br>
7) financial 16%<br>
8) healthinsurance 10%<br>
9) diapers 6%<br>
10) formula 5%<br>
10) food 4%<br>
13) housing 2%<br>
14) transportation 1%<br>

* support_none 7%


# Persona: Black

## Respondants
* 41

## Niegborhood

## Household Income
* 125,001 and over     17%
* 100,001 - 125,000    17%
* 75,001 - 100,000     17%
* 40,001 - 75,000      20%
* 40,000 and under     29%


## Perception of Wellness
* We've got this!      32%
* We're doing OK.      32%
* So-So.               17%
* It's a struggle.     15%
* Barely making it.    5%

## Perception of Financial Wellness
* We're doing OK.      32%
* We've got this!      32%
* So-So.               17%
* It's a struggle.     15%
* Barely making it.     5%

## Hierarachy of Needs
1) social 66%<br>
2) mentalhealth 49%<br>
3) lactation 44%<br>
4) postpartumdoula 37%<br>
7) financial 32%<br>
5) birthdoula 22%<br>
9) diapers 22%<br>
10) food 20%<br>
6) covid19education 17%<br>
8) healthinsurance 12%<br>
11) formula 12%<br>
12) housing 7%<br>
13) transportation 2%<br>

* support_none 0.0

# Persona: Asian

## Respondants
* 36

## Niegborhood

## Household Income
* 125,001 and over     50%
* 100,001 - 125,000    6%
* 75,001 - 100,000     17%
* 40,001 - 75,000      22%
* 40,000 and under     3%
* undisclosed          3%


## Perception of Wellness
* We've got this!      14%
* We're doing OK.      53%
* So-So.               19%
* It's a struggle.     11%
* Barely making it.    3%


## Perception of Financial Wellness
* We've got this!      31%
* We're doing OK.      44%
* So-So.               17%
* It's a struggle.     6%
* Barely making it.    3%

## Hierarachy of Needs
1) social 75%<br>
2) lactation 53%<br>
3) mentalhealth 47%<br>
4) postpartumdoula 47%<br>
5) financial 33%<br>
6) covid19education 28%<br>
7) birthdoula 25%<br>
8) healthinsurance 25%<br>
9) food 22%<br>
10) diapers 14%<br>
11) formula 14%<br>
12) housing 6%<br>
13) transportation 6%<br>

* support_none 8%

# Persona: Non_White

## Respondants
* 110

## Niegborhood

## Household Income
* 125,001 and over     32%
* 100,001 - 125,000    12%
* 75,001 - 100,000     19%
* 40,001 - 75,000      19%
* 40,000 and under     17%


## Perception of Wellness
* We've got this!      17%
* We're doing OK.      51%
* So-So.               17%
* It's a struggle.     12%
* Barely making it.    2%


## Perception of Financial Wellness
* We've got this!      36%
* We're doing OK.      35%
* So-So.               17%
* It's a struggle.     10%
* Barely making it.    3%


## Hierarachy of Needs
1) social 72%<br>
2) lactation 52%<br>
3) mentalhealth 51%<br>
4) postpartumdoula 45%<br>
5) financial 33%<br>
6) birthdoula 26%<br>
7) covid19education 23%<br>
8) diapers 22%<br>
9) food 22%<br>
10) healthinsurance 19%<br>
11) formula 15%<br>
12) housing 9%<br>
13) transportation 5%<br>

* support_none 4%

# Looking at Personas by Ethnicity 

### There are enough large enough samples or hispanic and non-hispanic respondants to proceed (cut off is 25)

In [32]:
df.q08_hispanic.value_counts()

No               585
Zero Response     87
Yes               40
Name: q08_hispanic, dtype: int64

In [34]:
# preparing data frames

df_hispanic = df[df.q08_hispanic == "Yes"]
df_nonhispanic = df[df.q08_hispanic == "No"]

print(df_hispanic.q08_hispanic.value_counts())
print('~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~`')
print(df_nonhispanic.q08_hispanic.value_counts())

Yes    40
Name: q08_hispanic, dtype: int64
~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~`
No    585
Name: q08_hispanic, dtype: int64


In [35]:
# list of dataframes and titles
df_list = [df_hispanic,df_nonhispanic]
title_list = ['Hispanic','Non_Hispanic']

## Respondants

In [37]:
num = 0
for df1 in df_list:
    
    print(title_list[num] + ": Number of Respondants")
    print(len(df1))
    print('~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~')
    num += 1

Hispanic: Number of Respondants
40
~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
Non_Hispanic: Number of Respondants
585
~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~


## Neighborhood

## household Income

In [44]:
num = 0
for df1 in df_list:
    
    print(title_list[num] + ": Income Distribution")
    print(df1.income.value_counts(normalize=True))
    print('~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~')
    num += 1

Hispanic: Income Distribution
125,001 and over     0.350
100,001 - 125,000    0.250
40,000 and under     0.175
75,001 - 100,000     0.150
40,001 - 75,000      0.050
nan                  0.025
Name: income, dtype: float64
~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
Non_Hispanic: Income Distribution
125,001 and over     0.382906
75,001 - 100,000     0.186325
40,001 - 75,000      0.174359
100,001 - 125,000    0.162393
40,000 and under     0.080342
nan                  0.013675
Name: income, dtype: float64
~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~


## Perceived Wellness

In [41]:
num = 0
for df1 in df_list:
    
    print(title_list[num] + ": Wellness Distribution")
    print(df1.q01_overall.value_counts(normalize=True))
    print('~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~')
    num += 1

Hispanic: Wellness Distribution
We're doing OK.     0.600
We've got this!     0.200
So-So.              0.175
It's a struggle.    0.025
Name: q01_overall, dtype: float64
~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
Non_Hispanic: Wellness Distribution
We're doing OK.      0.464957
So-So.               0.223932
It's a struggle.     0.157265
We've got this!      0.143590
Barely making it.    0.010256
Name: q01_overall, dtype: float64
~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~


## Perceved Financial Wellness

In [42]:
num = 0
for df1 in df_list:
    
    print(title_list[num] + ": Perception of Financial Wellness")
    print(df1.q02_financial.value_counts(normalize=True))
    print('~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~')
    num += 1

Hispanic: Perception of Financial Wellness
We've got this!      0.425
We're doing OK.      0.375
It's a struggle.     0.125
So-So.               0.050
Barely making it.    0.025
Name: q02_financial, dtype: float64
~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
Non_Hispanic: Perception of Financial Wellness
We've got this!      0.461538
We're doing OK.      0.340171
So-So.               0.111111
It's a struggle.     0.068376
Barely making it.    0.018803
Name: q02_financial, dtype: float64
~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~


## higharcy of needs by percentage

In [43]:
col_list = ['q06_social', 'q06_mentalhealth', 'q06_lactation','q06_postpartumdoula','q06_birthdoula',
            'q06_covid19education', 'q06_financial', 'q06_healthinsurance', 'q06_diapers', 'q06_food',
            'q06_formula', 'q06_support_none', 'q06_housing', 'q06_transportation']

index = 0 

for df1 in df_list:
    
    print(title_list[index] + ": Percent of Respondance who Identified Need by Rank")
    
    num = 1

    for col in col_list:

        print(f'{num})',col[4:],round(df1[df1[col] != "Zero Response"][col].count()/len(df1),2))

        num += 1
        
    print('~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~')
    print()
    
    index += 1

Hispanic: Percent of Respondance who Identified Need by Rank
1) social 0.72
2) mentalhealth 0.55
3) lactation 0.45
4) postpartumdoula 0.35
5) birthdoula 0.15
6) covid19education 0.28
7) financial 0.28
8) healthinsurance 0.15
9) diapers 0.15
10) food 0.15
11) formula 0.08
12) support_none 0.05
13) housing 0.05
14) transportation 0.05
~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~

Non_Hispanic: Percent of Respondance who Identified Need by Rank
1) social 0.78
2) mentalhealth 0.6
3) lactation 0.43
4) postpartumdoula 0.41
5) birthdoula 0.25
6) covid19education 0.24
7) financial 0.18
8) healthinsurance 0.11
9) diapers 0.08
10) food 0.07
11) formula 0.07
12) support_none 0.06
13) housing 0.03
14) transportation 0.01
~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~



# Persona: Hispanic


## Respondants
* 40


## Niegborhood


## Household Income
* 125,001 and over     35%
* 100,001 - 125,000    25%
* 75,001 - 100,000     15%
* 40,000 and under     18%
* 40,001 - 75,000      5%
* undisclosed          3%

## Perception of Wellness
* We're doing OK.     60%
* We've got this!     20%
* So-So.              18%
* It's a struggle.    3%
* Barely making it.   0%


## Perception of Financial Wellness
* We've got this!      43%
* We're doing OK.      38%
* It's a struggle.     13%
* So-So.               5%
* Barely making it.    3%


## Hierarachy of Needs
1) social 72%<br>
2) mentalhealth 55%<br>
3) lactation 45%<br>
4) postpartumdoula 35%<br>
5) covid19education 28%<br>
6) financial 28%<br>
7) birthdoula 15%<br>
8) healthinsurance 15%<br>
9) diapers 15%<br>
10) food 15%<br>
11) formula 8%<br>
12) housing 5%<br>
13) transportation 5%<br>

* support_none 5%

# Persona: Non_Hispanic

## Respondants
* 585


## Niegborhood

## Household Income
* 125,001 and over     38%
* 100,001 - 125,000    16%
* 75,001 - 100,000     19%
* 40,001 - 75,000      17%
* 40,000 and under     8%
* undisclosed          1%

## Perception of Wellness
* We're doing OK.      46%
* So-So.               22%
* It's a struggle.     16%
* We've got this!      14&
* Barely making it.    1%

## Perception of Financial Wellness
* We've got this!      46%
* We're doing OK.      34%
* So-So.               11%
* It's a struggle.     7%
* Barely making it.    2%

## Hierarachy of Needs
1) social 78%<br>
2) mentalhealth 60%<br>
3) lactation 43%<br>
4) postpartumdoula 41%<br>
5) birthdoula 25%<br>
6) covid19education 24%<br>
7) financial 18%<br>
8) healthinsurance 11%<br>
9) diapers 8%<br>
10) food 7%<br>
11) formula 7%<br>
12) housing 3%<br>
13) transportation 1%<br>

* support_none 6%

# Looking at persona's by Household Income

In [45]:
df.income.value_counts()

125,001 and over     238
75,001 - 100,000     115
100,001 - 125,000    105
40,001 - 75,000      104
nan                   96
40,000 and under      54
Name: income, dtype: int64

In [46]:
# creating data frames
df_over125 = df[df.income == "125,001 and over"]
df_100_125 = df[df.income == "100,001 - 125,000"]
df_75_100 = df[df.income == "75,001 - 100,000"]
df_40_75 = df[df.income == "40,001 - 75,000"]
df_under40 = df[df.income == "40,000 and under"]

In [48]:
# list of dataframes and titles
df_list = [df_over125, df_100_125, df_75_100, df_40_75, df_under40]
title_list = ["125,001 and over", "100,001 - 125,000", "75,001 - 100,000", "40,001 - 75,000", "40,000 and under"]

In [51]:
index = 0

for df1 in df_list:
    
    print(title_list[index] + " Value Counts:")
    print(df1.income.value_counts())
    print('~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~')
    index += 1
    

125,001 and over Value Counts:
125,001 and over    238
Name: income, dtype: int64
~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
100,001 - 125,000 Value Counts:
100,001 - 125,000    105
Name: income, dtype: int64
~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
75,001 - 100,000 Value Counts:
75,001 - 100,000    115
Name: income, dtype: int64
~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
40,001 - 75,000 Value Counts:
40,001 - 75,000    104
Name: income, dtype: int64
~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
40,000 and under Value Counts:
40,000 and under    54
Name: income, dtype: int64
~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~


## Neigborhood

## Perceved Wellness

In [60]:
num = 0
for df1 in df_list:
    
    print(title_list[num] + ": Wellness Distribution")
    print(df1.q01_overall.value_counts(normalize=True))
    print('~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~')
    num += 1

125,001 and over: Wellness Distribution
We're doing OK.      0.508403
So-So.               0.222689
We've got this!      0.138655
It's a struggle.     0.121849
Barely making it.    0.008403
Name: q01_overall, dtype: float64
~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
100,001 - 125,000: Wellness Distribution
We're doing OK.      0.495238
We've got this!      0.200000
So-So.               0.190476
It's a struggle.     0.104762
Barely making it.    0.009524
Name: q01_overall, dtype: float64
~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
75,001 - 100,000: Wellness Distribution
We're doing OK.     0.486957
So-So.              0.200000
It's a struggle.    0.173913
We've got this!     0.139130
Name: q01_overall, dtype: float64
~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
40,001 - 75,000: Wellness Distribution
We're doing OK.      0.403846
So-So.               0.259615
It's a struggle.     0.201923
We've got this!      0.12

In [55]:
## Peceved Financial Wellness

In [61]:
num = 0
for df1 in df_list:
    
    print(title_list[num] + ": Perception of Financial Wellness")
    print(df1.q02_financial.value_counts(normalize=True))
    print('~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~')
    num += 1

125,001 and over: Perception of Financial Wellness
We've got this!     0.684874
We're doing OK.     0.260504
So-So.              0.050420
It's a struggle.    0.004202
Name: q02_financial, dtype: float64
~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
100,001 - 125,000: Perception of Financial Wellness
We've got this!     0.533333
We're doing OK.     0.361905
So-So.              0.085714
It's a struggle.    0.019048
Name: q02_financial, dtype: float64
~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
75,001 - 100,000: Perception of Financial Wellness
We're doing OK.      0.486957
We've got this!      0.330435
So-So.               0.113043
It's a struggle.     0.052174
Barely making it.    0.017391
Name: q02_financial, dtype: float64
~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
40,001 - 75,000: Perception of Financial Wellness
We're doing OK.      0.394231
We've got this!      0.211538
So-So.               0.182692
It's a st

## Hiarchy of Needs

In [62]:
col_list = ['q06_social', 'q06_mentalhealth', 'q06_lactation','q06_postpartumdoula','q06_birthdoula',
            'q06_covid19education', 'q06_financial', 'q06_healthinsurance', 'q06_diapers', 'q06_food',
            'q06_formula', 'q06_support_none', 'q06_housing', 'q06_transportation']

index = 0 

for df1 in df_list:
    
    print(title_list[index] + ": Percent of Respondance who Identified Need by Rank")
    
    num = 1

    for col in col_list:

        print(f'{num})',col[4:],round(df1[df1[col] != "Zero Response"][col].count()/len(df1),2))

        num += 1
        
    print('~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~')
    print()
    
    index += 1

125,001 and over: Percent of Respondance who Identified Need by Rank
1) social 0.83
2) mentalhealth 0.59
3) lactation 0.43
4) postpartumdoula 0.48
5) birthdoula 0.27
6) covid19education 0.32
7) financial 0.06
8) healthinsurance 0.08
9) diapers 0.01
10) food 0.02
11) formula 0.03
12) support_none 0.04
13) housing 0.01
14) transportation 0.0
~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~

100,001 - 125,000: Percent of Respondance who Identified Need by Rank
1) social 0.77
2) mentalhealth 0.64
3) lactation 0.48
4) postpartumdoula 0.35
5) birthdoula 0.24
6) covid19education 0.27
7) financial 0.09
8) healthinsurance 0.1
9) diapers 0.06
10) food 0.04
11) formula 0.05
12) support_none 0.09
13) housing 0.02
14) transportation 0.01
~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~

75,001 - 100,000: Percent of Respondance who Identified Need by Rank
1) social 0.79
2) mentalhealth 0.58
3) lactation 0.52
4) postpartumdoula 0.4
5

# Persona: 125,001 and over

## Respondants
* 238

## Niegborhood


## Perception of Wellness

* We've got this!      14%
* We're doing OK.      51%
* So-So.               22%
* It's a struggle.     12%
* Barely making it.    1%

## Perception of Financial Wellness

* We've got this!     68%
* We're doing OK.     26%
* So-So.               5%
* It's a struggle.    <1%
* Barely making it.    0%

## Hierarachy of Needs

1) social 83% <br>
2) mentalhealth 59%<br>
3) postpartumdoula 48%<br>
4) lactation 43%<br>
5) covid19education 32%<br>
6) birthdoula 27%<br>
7) healthinsurance 8%<br>
8) financial 6%<br>
9) formula 3%<br>
10) food 2%<br>
11) diapers 1%<br>
12) housing 1%<br>
13) transportation 0%<br>

* support_none 4%<br>

# Persona: 100,001 - 125,000

## Respondants
* 105

## Niegborhood


## Perception of Wellness

* We're doing OK.      0.495238
* We've got this!      0.200000
* So-So.               0.190476
* It's a struggle.     0.104762
* Barely making it.    0.009524

## Perception of Financial Wellness
* We've got this!     53%
* We're doing OK.     36%
* So-So.              9%
* It's a struggle.    2%
* Barely making it.   0%

## Hierarachy of Needs
1) social 77%<br>
2) mentalhealth 64%<br>
3) lactation 48%<br>
4) postpartumdoula 35%<br>
5) covid19education 27%<br>
6) birthdoula 24%<br>
7) healthinsurance 10%<br>
8) financial 9%<br>
9) diapers 6%<br>
10) formula 5%<br>
11) food 4%<br>
12) housing 2%<br>
13) transportation 1%<br>

* support_none 9%

# Persona: 75,001 - 100,000

## Respondants
* 115

## Niegborhood


## Perception of Wellness
* We've got this!     14%
* We're doing OK.     49%
* So-So.              20%
* It's a struggle.    17%
* Barely making it.    0%


## Perception of Financial Wellness
* We're doing OK.      49%
* We've got this!      33%
* So-So.               11%
* It's a struggle.     5%
* Barely making it.    2%


## Hierarachy of Needs
1) social 79%<br>
2) mentalhealth 58%<br>
3) lactation 52%<br>
4) postpartumdoula 40%<br>
5) birthdoula 23%<br>
6) covid19education 19%<br>
7) financial 18%<br>
8) healthinsurance 10%<br>
9) diapers 8%<br>
10) formula 5%<br>
11) food 4%<br>
12) housing 4%<br>
13) transportation 3%<br>

* support_none 7%

# Persona: 40,001 - 75,000

## Respondants
* 104

## Niegborhood


## Perception of Wellness
* We're doing OK.      40%
* So-So.               26%
* It's a struggle.     20%
* We've got this!      13%
* Barely making it.    1%


## Perception of Financial Wellness
* We're doing OK.      39%
* We've got this!      21%
* So-So.               18%
* It's a struggle.     18%
* Barely making it.    3%


## Hierarachy of Needs
1) social 75%<br>
2) mentalhealth 62%<br>
3) financial 40%<br>
4) postpartumdoula 38%<br>
5) lactation 35%<br>
6) birthdoula 24%<br>
7) diapers 19%<br>
8) covid19education 15%<br>
9) food 15%<br>
10) formula 13%<br>
11) healthinsurance 12%<br>
12) housing 3%<br>
14) transportation 2%<br>

* support_none 9%<br>

# Persona: 40,000 and Under

## Respondants
* 54

## Niegborhood

## Perception of Wellness
* We're doing OK.      35%
* So-So.               28%
* It's a struggle.     17%
* We've got this!      17%
* Barely making it.    4%

## Perception of Financial Wellness
* It's a struggle.     31%
* So-So.               26%
* We're doing OK.      26%
* Barely making it.    9%
* We've got this!      7%


## Hierarachy of Needs
1) social 56%<br>
2) mentalhealth 56%<br>
3) financial 52%<br>
4) lactation 33%<br>
5) healthinsurance 31%<br>
6) diapers 30%<br>
7) postpartumdoula 26%<br>
8) food 24%<br>
9) birthdoula 22%<br>
10) formula 22%<br>
11) housing 13%<br>
12) covid19education 11%<br>
13) transportation 4%<br>

* support_none 4%

# Higharchy by Stage

### Excluding 1st trimester and caregiver as sample size is not large enough.

In [63]:
df.pregnancy_stage.value_counts()

1 year           416
3rd trimester     90
2 years           74
2nd trimester     62
multi             46
1st trimester     14
caregiver         10
Name: pregnancy_stage, dtype: int64

In [66]:
# setting up data frames
df_1y = df[df.pregnancy_stage == "1 year"]
df_2y = df[df.pregnancy_stage == "2 years"]
df_2t = df[df.pregnancy_stage == "2nd trimester"]
df_3t = df[df.pregnancy_stage == "3rd trimester"]
df_multi = df[df.pregnancy_stage == "multi"]

In [69]:
df_list = [df_1y,df_2y,df_2t,df_3t,df_multi]
title_list = ["1 year","2 years","2nd trimester","3rd trimester","multi"]

In [72]:
col_list = ['q06_social', 'q06_mentalhealth', 'q06_lactation','q06_postpartumdoula','q06_birthdoula',
            'q06_covid19education', 'q06_financial', 'q06_healthinsurance', 'q06_diapers', 'q06_food',
            'q06_formula', 'q06_support_none', 'q06_housing', 'q06_transportation']

index = 0 

for df1 in df_list:
    
    print(title_list[index] + ": Percent of Respondance who Identified Need")
    
    num = 1

    for col in col_list:

        print(f'{num})',col[4:],round(df1[df1[col] != "Zero Response"][col].count()/len(df1),2))

        num += 1
        
    print('~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~')
    print()
    
    index += 1

1 year: Percent of Respondance who Identified Need
1) social 0.72
2) mentalhealth 0.56
3) lactation 0.41
4) postpartumdoula 0.38
5) birthdoula 0.19
6) covid19education 0.21
7) financial 0.17
8) healthinsurance 0.09
9) diapers 0.07
10) food 0.06
11) formula 0.06
12) support_none 0.05
13) housing 0.03
14) transportation 0.01
~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~

2 years: Percent of Respondance who Identified Need
1) social 0.68
2) mentalhealth 0.61
3) lactation 0.12
4) postpartumdoula 0.15
5) birthdoula 0.08
6) covid19education 0.12
7) financial 0.22
8) healthinsurance 0.05
9) diapers 0.07
10) food 0.09
11) formula 0.08
12) support_none 0.04
13) housing 0.04
14) transportation 0.04
~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~

2nd trimester: Percent of Respondance who Identified Need
1) social 0.68
2) mentalhealth 0.45
3) lactation 0.5
4) postpartumdoula 0.5
5) birthdoula 0.42
6) covid19education 0.29
7) 